staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler_test.go

```diff
@@ -638,6 +648,217 @@ func testHandlerConversion(t *testing.T, enableWatchCache bool) {
 	}
 }
 
+func TestDecoder(t *testing.T) {
+	multiVersion := `
+	{
+	"apiVersion": "stable.example.com/v1beta1",
+	"kind": "MultiVersion",
+	"metadata": {
+		"name": "my-mv"
+	},
+	"num": 1,
+	"num": 2,
+	"unknown": "foo"
+	}
+	`
+	multiVersionYAML := `
+apiVersion: stable.example.com/v1beta1
+kind: MultiVersion
+metadata:
+  name: my-mv
+num: 1
+num: 2
+unknown: foo`
+
+	decoded := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	decoded.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "stable.example.com",
+		Version: "v1beta1",
+		Kind:    "MultiVersion",
+	})
+	decoded.SetName("my-mv")
+	decoded.SetGeneration(1)
+	unstructured.SetNestedField(decoded.Object, int64(2), "num")
+	unstructured.SetNestedField(decoded.Object, nil, "metadata", "creationTimestamp")
+
+	decodedPreserveUnknown := decoded.DeepCopy()
+	unstructured.SetNestedField(decodedPreserveUnknown.Object, "foo", "unknown")
+
+	testcases := []struct {
+		name                  string
+		body                  string
+		yaml                  bool
+		strictDecoding        bool
+		preserveUnknownFields bool
+		crd                   *v1.CustomResourceDefinition
```

Comment:
There are a lot of variables in that test, which really makes tests harder to comprehend. The CRD parameter isn't even changing from test to test, as far as I could tell, can we just make this into a const or hard-code it somewhere?